scheduler/scheduler_test.go

```diff
@@ -0,0 +1,167 @@
+package scheduler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/robfig/cron/v3"
+)
+
+const (
+	everySecond   = "* * * * * *"
+	maxRunTime    = 200 * time.Millisecond
+	scheduleLimit = 5 * time.Second
+)
+
+func TestScheduleTask(t *testing.T) {
+	s := NewWithSeconds()
+
+	// Add task to scheduler
+	jid, err := s.Add(everySecond, nil, maxRunTime, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var noEnt cron.Entry
+	ent := s.sched.Entry(jid)
+	if ent == noEnt {
+		t.Fatal("job was not scheduled")
+	}
+
+	// Test that job starts
+	var job *Job
+	select {
+	case job = <-s.RunChan():
+		if job.ID() != ent.ID {
+			t.Fatal("wrong ID, got", job.ID(), "expected", jid)
+		}
+		t.Log("started", job.ID(), "run", job.RunCount())
+	case <-time.After(time.Second + maxRunTime):
+		t.Fatal("timed out waiting for job to start")
+	}
+
+	if job.RunCount() != 1 {
+		t.Errorf("job has wrong run count, expected 1, got %d", job.RunCount())
+	}
+
+	// Test that job stops
+	select {
+	case <-job.RunContext().Done():
+		t.Log("stopped", job.ID(), "run", job.RunCount())
+	case <-time.After(time.Second + maxRunTime):
+		t.Fatal("timed out waiting for job to stop")
+	}
+
+	// Test that job starts again
+	select {
+	case job = <-s.RunChan():
+		if job.ID() != ent.ID {
+			t.Fatal("wrong ID, got", job.ID(), "expected", jid)
+		}
+		t.Log("started", job.ID(), "run", job.RunCount())
+	case <-time.After(time.Second + maxRunTime):
+		t.Fatal("timed out waiting for job to start")
+	}
+
+	if job.RunCount() != 2 {
+		t.Errorf("job has wrong run count, expected 2, got %d", job.RunCount())
+	}
+
+	job.End()
+
+	// Test that job stops immediately
+	select {
+	case <-job.RunContext().Done():
+		t.Log("stopped", job.ID(), "run", job.RunCount())
+	case <-time.After(time.Millisecond):
```

Comment:
I think these `1ms` timeouts might give us trouble, especially in CI where machines are often shared with other concurrent jobs. Something like 100ms or 200ms would be much safer.

Reply:
Done.  This particular wait is not even needed.